src/components/forms/ProxyDeviceForm.tsx

```diff
@@ -0,0 +1,364 @@
+import { FC } from "react";
+import {
+  Button,
+  Icon,
+  Input,
+  Label,
+  Select,
+  useNotify,
+} from "@canonical/react-components";
+import { useQuery } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { LxdProxyDevice } from "types/device";
+import { InstanceAndProfileFormikProps } from "./instanceAndProfileFormValues";
+import { fetchProfiles } from "api/profiles";
+import { getInheritedProxies } from "util/configInheritance";
+import Loader from "components/Loader";
+import ScrollableForm from "components/ScrollableForm";
+import RenameDeviceInput from "components/forms/RenameDeviceInput";
+import ConfigurationTable from "components/ConfigurationTable";
+import { MainTableRow } from "@canonical/react-components/dist/components/MainTable/MainTable";
+import { getConfigurationRowBase } from "components/ConfigurationRow";
+import classnames from "classnames";
+import {
+  addNoneDevice,
+  deduplicateName,
+  findNoneDeviceIndex,
+  removeDevice,
+} from "util/formDevices";
+import { getInheritedDeviceRow } from "components/forms/InheritedDeviceRow";
+import { deviceKeyToLabel } from "util/devices";
+import { ensureEditMode } from "util/instanceEdit";
+import NewProxyBtn from "components/forms/NewProxyBtn";
+import ConfigFieldDescription from "pages/settings/ConfigFieldDescription";
+import { optionEnabledDisabled, optionYesNo } from "util/instanceOptions";
+
+interface Props {
+  formik: InstanceAndProfileFormikProps;
+  project: string;
+}
+
+const ProxyDeviceForm: FC<Props> = ({ formik, project }) => {
+  const notify = useNotify();
+
+  const {
+    data: profiles = [],
+    isLoading: isProfileLoading,
+    error: profileError,
+  } = useQuery({
+    queryKey: [queryKeys.profiles],
+    queryFn: () => fetchProfiles(project),
+  });
+
+  if (profileError) {
+    notify.failure("Loading profiles failed", profileError);
+  }
+
+  const inheritedProxies = getInheritedProxies(formik.values, profiles);
+
+  const existingDeviceNames: string[] = [];
+  existingDeviceNames.push(...inheritedProxies.map((item) => item.key));
+  existingDeviceNames.push(...formik.values.devices.map((item) => item.name));
+
+  const addProxy = () => {
+    const copy = [...formik.values.devices];
+    copy.push({
+      type: "proxy",
+      name: deduplicateName("proxy", 1, existingDeviceNames),
+    });
+    void formik.setFieldValue("devices", copy);
+  };
+
+  const hasCustomProxy = formik.values.devices.some(
+    (item) => item.type === "proxy",
+  );
+
+  console.log(formik.values.devices);
+
+  const getProxyDeviceFormRows = (
+    label: string,
+    fieldName: string,
+    index: number,
+    options: {
+      label: string;
+      value: string;
+      disabled?: boolean;
+    }[],
+    help?: string,
+    value?: string,
```

Comment:
```suggestion
    value?: string,
    help?: string,
```
Best to switch them around, so we don't have to pass undefined below when no help text is defined.